src/_imaging.c

```diff
@@ -4071,14 +4091,20 @@ _set_blocks_max(PyObject *self, PyObject *args) {
     if (blocks_max < 0) {
         PyErr_SetString(PyExc_ValueError, "blocks_max should be greater than 0");
         return NULL;
-    } else if (
-        (unsigned long)blocks_max >
-        SIZE_MAX / sizeof(ImagingDefaultArena.blocks_pool[0])) {
+    }
+
+    MUTEX_LOCK(&ImagingDefaultArena.mutex);
+    size_t blocksize = sizeof(ImagingDefaultArena.blocks_pool[0]);
```

Comment:
```c
sizeof(ImagingDefaultArena.blocks_pool[0])
````
is evaluated at compile time so there is no need to lock here.

In fact, you could even use
```c
sizeof(((ImagingMemoryArena*)0)->blocks_pool[0])
```

Reply:
Thanks for the pointer! Updated.